xarray/core/utils.py

```diff
@@ -533,3 +533,41 @@ def ensure_us_time_resolution(val):
     elif np.issubdtype(val.dtype, np.timedelta64):
         val = val.astype('timedelta64[us]')
     return val
+
+
+class HiddenKeyDict(MutableMapping):
+    '''
+    Acts like a normal dictionary, but hides certain keys.
+    '''
+    # ``__init__`` method required to create instance from class.
+    def __init__(self, data, hidden_keys):
+        self._data = data
+        if type(hidden_keys) is not list:
+            hidden_keys = [ hidden_keys ]
```

Comment:
Personally, I would drop this clause and always require a list by raising an error instead. There's little too be gained by making internal classes more flexible and less type safe.